packages/navie/src/services/completion-service.ts

```diff
@@ -83,3 +83,29 @@ export function convertToMessage(message: Message): BaseMessage {
   }
   return new Cons({ content });
 }
+
+// Number of retries to attempt a completion before giving up
+export const CompletionRetries = (() => {
+  const env = process.env.APPMAP_NAVIE_COMPLETION_RETRIES;
+  if (env) {
+    const value = parseInt(env, 10);
+    if (Number.isInteger(value) && value > 0) {
```

Comment:
Or, just `if (value)`? In general these two functions are quite verbose for no good reason (that I can see). How about `export const CompletionRetries = parseInt(process.env.APPMAP_NAVIE_COMPLETION_RETRIES) || 5`?

Reply:
If the user's intent is to disable retries, `parseInt(...) || 5` doesn't allow the value to be set to zero. Similarly, `parseInt(...) ?? 5` allows the value to be set to `NaN`.

This function is slightly verbose, but it catches these corner cases.